templating/template.go

```diff
@@ -24,3 +28,21 @@ func (t *Template) GetData() []byte {
 	}
 	return t.Data
 }
+
+// NewTemplate creates a new template with the specified name and data.
+// It will strip out any commented lines from data, i.e. lines beginning with #.
+func NewTemplate(name string, data []byte) *Template {
+	ret := []string{}
+	lines := strings.Split(string(data), "\n")
+	for _, line := range lines {
+		tLine := strings.TrimSpace(line)
+		if !strings.HasPrefix(tLine, "#") {
+			// Append the original line to preserve any spacing.
+			ret = append(ret, line)
+		}
+	}
+	return &Template{
+		Name: name,
+		Data: []byte(strings.Join(ret, "\n")),
```

Comment:
Is the ending "\n" required?

Reply:
Not sure I understand - we split by `\n` so we're just adding them back in at this point.

Reply:
Oh, I see.